downloads/swan-lake-release-notes/swan-lake-preview1/RELEASE_NOTE.md

```diff
@@ -264,24 +264,39 @@ public function main() {
 }
 ```
 
-### The Never type
-The `never` type represents the type of values that never occur. This can be useful to describe the return type of a function if the function never returns. No value can ever belong to the `never` type. Thus, it can not be declared or a value cannot be assigned to it.
+### Never type
 
+The `never` type provides a way to describe the type that contains no shapes. 
+
+The `never` type is useful as the return type of a function that never terminates normally. 
 ```ballerina
-function somefunction(int age) returns never {
-   if (age < 18) {
-       error e = error("Invalid Age");
-       panic e;
-   }
+function somefunction() returns never {
+    panic error("Invalid");
 }
 ```
-Never types can be used to specify key-less tables. Here, the key constraint is type `never`
+The `never` type is a subtype of `nil`. Therefore, functions with the `never` type as the return type can only be invoked in call statements. i.e., they can never be called as an expression nor can the result of calling such a function be assigned to a variable. Since the `never` type represents the type which has no values, variables of type `never` cannot be declared.
```

Comment:
IMO saying therefore in the second sentence is misleading because we do allow assigning the result of calling a function returning nil to a variable.

```ballerina
   () x = foo();
```

Shall we say
```suggestion
The `never` type is a subtype of `nil`. Functions with the `never` type as the return type can only be invoked in call statements. i.e., they can never be called as an expression nor can the result of calling such a function be assigned to a variable. Since the `never` type represents the type which has no values, variables of type `never` cannot be declared.
```

Reply:
+1

Reply:
+1

Reply:
We don't allow this one 
```ballerina
() x = foo();
```

Reply:
However, change is okay

Reply:
I think we do, there's no reason to disallow it actually.

Check

```balllerina
public function main() {
    () x = foo();
}

function foo() {
    
}
```